### PR TITLE
Fix #if block highlighting and POSIX types

### PR DESCRIPTION
**Pull request by sWski**

### CODE REVIEW

**Comment by 50Wliu** [comment]
Can you please add specs to this?  I'm aware that there's not much spec coverage right now, but every bit helps :).

**Comment by sWski** [comment]
Sure, I'll see what I can do :)

**Comment by sWski** [comment]
I have added specs for my changes and a little bit extended the existing ones. I hope they are sufficient, since Atom development is a new thing to me :)

File: spec/c-spec.coffee [hunk 1]
```diff
@@ -24,6 +24,41 @@ describe 'Language-C', ->
...
+      expect(lines[0][3]).toEqual value: '(', scopes: ["source.c", "meta.function.c", "meta.parens.c", "punctuation.section.parens.begin.c"]
+      expect(lines[0][6]).toEqual value: '{', scopes: ["source.c", "meta.function.c", "meta.block.c", "punctuation.section.block.begin.c"]
+
+    it 'tokenizes if-true-else preprocessor blocks', ->
+      lines = grammar.tokenizeLines '''
+        #if 1
+        int something() {
+          return 0;
+        }
+        #else
```
Comment (50Wliu) [inline spec/c-spec.coffee]: Can we also test that this else block is getting tokenized correctly?

File: spec/c-spec.coffee [hunk 2]
```diff
@@ -24,6 +24,41 @@ describe 'Language-C', ->
...
+        int something() {
+          return 0;
+        }
+        #endif
```
Comment (50Wliu) [inline spec/c-spec.coffee]: Ditto for this `endif`

File: grammars/c.cson [hunk 3]
```diff
@@ -450,9 +450,9 @@
         ]
       }
       {
-        'begin': ''
+        'begin': '\\G'
         'end': '(?=^\\s*#\\s*(else|endif)\\b)'
-        'name': 'comment.block.preprocessor.if-branch'
+        'contentName': 'comment.block.preprocessor.if-branch'
```
Comment (50Wliu) [inline grammars/c.cson]: Should there also be test coverage for this?

File: grammars/c.cson [hunk 4]
```diff
@@ -489,9 +489,9 @@
         ]
       }
       {
-        'begin': ''
+        'begin': '\\G'
         'end': '(?=^\\s*#\\s*(else|endif)\\b)'
-        'name': 'comment.block.preprocessor.if-branch.in-block'
+        'contentName': 'comment.block.preprocessor.if-branch.in-block'
```
Comment (50Wliu) [inline grammars/c.cson]: And this?

Comment (50Wliu) [inline grammars/c.cson]: And this?

**Comment by MaximSokolov** [comment]
@50Wliu Specs have been added

**Comment by 50Wliu** [comment]
@sWski Sorry for the delay in merging this!

(For next time: it would be great if you could comment after addressing comments so that we get a notification)

**Pull request merged**